### PR TITLE
feat(update transaction): implement update transaction method

### PR DESCRIPTION
**Pull request by HoangLiarPhuongLe**
- Implement update transaction method

### CODE REVIEW

File: src/javascripts/constants/message.js [hunk 1]
```diff
@@ -14,5 +14,6 @@ export const ERROR_MESSAGE = {
 }
 
 export const ADD_TRANSACTION_MESSAGE = {
-    ADD_TRANSACTION_SUCCESS: "Add Transaction Succcess !"
+    ADD_TRANSACTION_SUCCESS: "Add Transaction Succcess !",
```
Comment (phuocduong-agilityio) [inline src/javascripts/constants/message.js]: Should be "Text content!" not "Text content !"

File: src/javascripts/controllers/transactionController.js [hunk 2]
```diff
@@ -30,8 +32,14 @@ class TransactionController {
     }
 
     saveTransaction = async(data) => {
-        await this.service.transactionService.addTransaction(data);
-        this.view.snackbar.showSnackBar('success', ADD_TRANSACTION_MESSAGE.ADD_TRANSACTION_SUCCESS);
+        if(!data.id){
+            await this.service.transactionService.addTransaction(data);
+            this.view.snackbar.showSnackBar('success', ADD_TRANSACTION_MESSAGE.ADD_TRANSACTION_SUCCESS);
+        }else{
```
Comment (phuocduong-agilityio) [inline src/javascripts/controllers/transactionController.js]: Format

File: src/javascripts/services/transactionService.js [hunk 3]
```diff
@@ -36,6 +35,7 @@ class TransactionService {
     getTransactionById = async(id) => {
         const data = await this.api.getTransactionById(id);
         const transaction = new Transaction(data);
+        console.log(transaction);
```
Comment (phuocduong-agilityio) [inline src/javascripts/services/transactionService.js]: Remove console

File: src/javascripts/views/homeView.js [hunk 4]
```diff
@@ -56,7 +55,7 @@ class HomeView {
        const categoryTemplate = TransactionTemplate.createCategory(category, transactions, totalOutFlow);
 
        this.categoryListEl.innerHTML += categoryTemplate;
-
+
```
Comment (phuocduong-agilityio) [inline src/javascripts/views/homeView.js]: Remove line here

File: src/javascripts/views/homeView.js [hunk 5]
```diff
@@ -77,7 +76,19 @@ class HomeView {
         this.summaryTabEl.innerHTML = summaryTemplate;
     }
 
+    
     //----- EVENT HANDLER -----//
+    addDelegateShowInfo = (showInfo) => {
```
Comment (phuocduong-agilityio) [inline src/javascripts/views/homeView.js]: showInfor -> Choose another name with prefix/post fix func/handler/etc.

File: src/javascripts/constants/message.js (hunk 1 above)
Comment (HoangLiarPhuongLe) [reply src/javascripts/constants/message.js]: I fixed @phuocduong-agilityio . Please help me check. Thanks!

File: src/javascripts/controllers/transactionController.js (hunk 2 above)
Comment (HoangLiarPhuongLe) [reply src/javascripts/controllers/transactionController.js]: I fixed @phuocduong-agilityio . Please help me check. Thanks!

File: src/javascripts/services/transactionService.js (hunk 3 above)
Comment (HoangLiarPhuongLe) [reply src/javascripts/services/transactionService.js]: I fixed @phuocduong-agilityio . Please help me check. Thanks!

File: src/javascripts/views/homeView.js (hunk 4 above)
Comment (HoangLiarPhuongLe) [reply src/javascripts/views/homeView.js]: I fixed @phuocduong-agilityio . Please help me check. Thanks!

File: src/javascripts/views/homeView.js (hunk 5 above)
Comment (HoangLiarPhuongLe) [reply src/javascripts/views/homeView.js]: I fixed @phuocduong-agilityio . Please help me check. Thanks!

File: src/javascripts/services/apiService.js [hunk 6]
```diff
@@ -54,7 +56,7 @@ class ApiService{
     
       try {
         // Make a POST request to the API endpoint with the transaction object as the request body
-        const response = await fetch(`${this.baseUrl}${this.path}`, {
+        const response = await fetch(`${this.baseUrl}${this.path}/${transaction.id}`, {
```
Comment (phuocduong-agilityio) [inline src/javascripts/services/apiService.js]: Add empty line after const/let block

File: src/javascripts/services/transactionService.js [hunk 7]
```diff
@@ -48,6 +47,12 @@ class TransactionService {
     updateTransaction = async(data) => {
         const transaction = new Transaction(data);
         await this.api.updateTransaction(transaction);
+        this.transactionList = this.transactionList.map((item) => {
+            if (item.id == transaction.id) {
```
Comment (phuocduong-agilityio) [inline src/javascripts/services/transactionService.js]: Should not use ==, use === instead. For different types, pls use parse

File: src/javascripts/views/homeView.js [hunk 8]
```diff
@@ -77,7 +75,19 @@ class HomeView {
         this.summaryTabEl.innerHTML = summaryTemplate;
     }
 
+    
     //----- EVENT HANDLER -----//
+    addDelegateShowInfo = (handlerTransactionInfo) => {
+        this.transactionListEl = this.categoryListEl.querySelector(".transaction-list");
+        this.transactionListEl.addEventListener("click", (event) => {
+            const el = event.target.closest(".transaction-detail");
```
Comment (phuocduong-agilityio) [inline src/javascripts/views/homeView.js]: No need empty line between const.
Should define meaning variable.

File: src/javascripts/views/homeView.js [hunk 9]
```diff
@@ -77,7 +75,19 @@ class HomeView {
         this.summaryTabEl.innerHTML = summaryTemplate;
     }
 
+    
     //----- EVENT HANDLER -----//
+    addDelegateShowInfo = (handlerTransactionInfo) => {
```
Comment (phuocduong-agilityio) [inline src/javascripts/views/homeView.js]: Could you define more specific name that what handler use for?

File: src/javascripts/services/transactionService.js (hunk 7 above)
Comment (HoangLiarPhuongLe) [reply src/javascripts/services/transactionService.js]: I fixed @phuocduong-agilityio . Please help me check. Thanks!

File: src/javascripts/views/homeView.js (hunk 8 above)
Comment (HoangLiarPhuongLe) [reply src/javascripts/views/homeView.js]: I fixed @phuocduong-agilityio . Please help me check. Thanks!

File: src/javascripts/views/homeView.js (hunk 9 above)
Comment (HoangLiarPhuongLe) [reply src/javascripts/views/homeView.js]: I fixed @phuocduong-agilityio . Please help me check. Thanks!

**Pull request merged**